indico/modules/events/registration/api.py

```diff
@@ -7,19 +7,98 @@
 
 from flask import jsonify, request, session
 from sqlalchemy.orm import joinedload
+from webargs import fields
 from werkzeug.exceptions import BadRequest, Forbidden
 
 from indico.core import signals
+from indico.modules.events import Event
 from indico.modules.events.controllers.base import RHProtectedEventBase
-from indico.modules.events.models.events import Event
-from indico.modules.events.registration.models.registrations import RegistrationState
+from indico.modules.events.registration.models.forms import RegistrationForm
+from indico.modules.events.registration.models.registrations import Registration, RegistrationState
 from indico.modules.events.registration.util import build_registration_api_data, build_registrations_api_data
-from indico.web.rh import RH, oauth_scope
+from indico.web.args import use_kwargs
+from indico.web.rh import RH, cors, json_errors, oauth_scope
+
+
+@json_errors
+@cors
+@oauth_scope('registrants')
+class RHAPICheckinBase(RHProtectedEventBase):
+    """Base class for the Check-in API."""
+
+
+class RHAPIEvent(RHAPICheckinBase):
+    """Manage a single event."""
+
+    def _process(self):
+        from indico.modules.events.registration.schemas import CheckinEventSchema
+        return jsonify(CheckinEventSchema().dump(self.event))
+
+
+class RHAPIRegFormBase(RHAPICheckinBase):
+    """Base class for registration forms management."""
+
+    def _check_access(self):
+        RHAPICheckinBase._check_access(self)
+        if not self.event.can_manage(session.user, permission='registration'):
+            raise Forbidden()
+
+    def _process_args(self):
+        RHAPICheckinBase._process_args(self)
+        self.regform = (RegistrationForm.query
+                        .filter_by(id=request.view_args['reg_form_id'], is_deleted=False)
```

Comment:
the `is_deleted` check is included in the `with_parent` since the relationship contains it in its join criterion